apollo-client/src/main/java/com/ctrip/framework/apollo/spring/spi/DefaultApolloConfigRegistrarHelper.java

```diff
@@ -40,12 +48,24 @@ public void registerBeanDefinitions(AnnotationMetadata importingClassMetadata, B
         SpringValueProcessor.class);
     BeanRegistrationUtil.registerBeanDefinitionIfNotExists(registry, SpringValueDefinitionProcessor.class.getName(),
         SpringValueDefinitionProcessor.class);
-    BeanRegistrationUtil.registerBeanDefinitionIfNotExists(registry, ApolloJsonValueProcessor.class.getName(),
-        ApolloJsonValueProcessor.class);
+  }
+
+  private String[] resolveNamespaces(String[] namespaces) {
+    String[] resolvedNamespaces = new String[namespaces.length];
+    for (int i = 0; i < namespaces.length; i++) {
+      resolvedNamespaces[i] = this.configurableBeanFactory.resolveEmbeddedValue(namespaces[i]);
```

Comment:
Cannot resolve value here.

Because there are no `org.springframework.beans.factory.support.AbstractBeanFactory#embeddedValueResolvers` in `configurableBeanFactory` now.

May be `com.ctrip.framework.apollo.spring.annotation.ApolloConfigRegistrar` handle `@EnableApolloConfig` too early.

Reply:
When someone use `@FeignClient`, the following code will work
```java
@FeignClient(name = "${self.name}", url = "${self.web}")
public interface Client {

	@GetMapping(value = "/stores")
	String getStores();
}
```

`"${self.name}"` will be resolved.

When I check the source code, found that `FeignClientsRegistrar` use `environment` instead of `configurableBeanFactory`

The relative code in link
https://github.com/spring-cloud/spring-cloud-openfeign/blob/f9a8dd13f255b05ace30917048a6d12a6f7ad70a/spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/FeignClientsRegistrar.java#L261

May be we can use this way to resolve the value in `@EnableApolloConfig`

Reply:
We could give it a try

Reply:
Sure. It works.

The change in commit ee934e3c1ae3b3484de4259551a10d7176ea1f65,
and the test in commit cc1bbf7de43709bedc3f7f7c41ce8ae82b6f828e